twine/commands/upload.py

```diff
@@ -189,30 +190,46 @@ def upload(dists, repository, sign, identity, username, password, comment,
 
         }
 
+        md5_hash = hashlib.md5()
         with open(filename, "rb") as fp:
-            content = fp.read()
-            filedata = {
-                "content": (os.path.basename(filename), content),
-            }
-            data["md5_digest"] = hashlib.md5(content).hexdigest()
+            content = fp.read(4096)
+            while content:
+                md5_hash.update(content)
+                content = fp.read(4096)
+
+        data["md5_digest"] = md5_hash.hexdigest()
 
         signed_name = os.path.basename(filename) + ".asc"
         if signed_name in signatures:
             with open(signatures[signed_name], "rb") as gpg:
-                filedata["gpg_signature"] = (signed_name, gpg.read())
+                data["gpg_signature"] = (signed_name, gpg.read())
         elif sign:
             with open(filename + ".asc", "rb") as gpg:
-                filedata["gpg_signature"] = (signed_name, gpg.read())
+                data["gpg_signature"] = (signed_name, gpg.read())
 
         print("Uploading {0}".format(os.path.basename(filename)))
 
-        resp = session.post(
-            config["repository"],
-            data=dict((k, v) for k, v in data.items() if v),
-            files=filedata,
-            auth=(username, password),
-            allow_redirects=False,
-        )
+        data_to_send = []
+        for key, value in data.items():
+            if isinstance(value, (list, tuple)):
+                for item in value:
```

Comment:
otherwise contributors (and any other list/tuple) was being picked up [by this](https://github.com/sigmavirus24/requests-toolbelt/blob/master/requests_toolbelt/multipart/encoder.py#L218)